klog.go

```diff
@@ -1370,3 +1370,38 @@ func Exitf(format string, args ...interface{}) {
 	atomic.StoreUint32(&fatalNoStacks, 1)
 	logging.printf(fatalLog, logging.logr, format, args...)
 }
+
+// ObjectRef references a kubernetes object
+type ObjectRef struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace,omitempty"`
+}
+
+func (ref ObjectRef) String() string {
+	if ref.Namespace != "" {
+		return fmt.Sprintf("%s/%s", ref.Namespace, ref.Name)
+	}
+	return ref.Name
+}
+
+// NamespacedName is a subset of the metav1.Object interface
```

Comment:
Please say something like "this interface may expand in the future, but will always be a subset of the kubernetes metav1.Object interface" and maybe expand metav1 to a full path `k8s.io/apimachinery/pkg/apis/meta/v1`.  I want this to be a strong statement that whatever clients pass in should be a real Object and not some thing that just pretends to be Object.

Alternately, when we want to add UID we define a new interface and try casting to it.  e.g. https://play.golang.org/p/2X4qxzJz4Mn

@serathius - preference?

Reply:
I think expanding this interface is ok if we do a strong statement as Tim suggested. Defining new interfaces with optional functions would only obscure intention of using `kObj` only for real k8s Objects. 

We should add a comment that we reserve right to change this interface in any way that doesn't break compatibility to "meta.Object" interface